tools/devops/automation/scripts/RemoteMac.psm1

```diff
@@ -0,0 +1,65 @@
+<#
+    .SYNOPSIS
+        Executes a command on a remote machine using scp
+#>
+
+function Invoke-SshCommand {
+    param (
+        [Parameter(Mandatory)]
+        [string]
+        $RemoteHost,
+
+        [Parameter(Mandatory)]
+        [string]
+        $RemoteUserName,
+
+        [Parameter(Mandatory, ValueFromRemainingArguments)]
+        [string[]]
+        $CommandArguments
+    )
+
+    $cmd = "ssh -v -i `"$Env:ID_RSA_PATH`" -o StrictHostKeyChecking=no `"$($RemoteUserName)@$($RemoteHost)`" $CommandArguments"
+    Write-Host "Executing: $cmd"
+
+    ssh -v -i "$Env:ID_RSA_PATH" -o StrictHostKeyChecking=no "$($RemoteUserName)@$($RemoteHost)" @CommandArguments
+
+    if ($LastExitCode -ne 0) {
+        throw [System.Exception]::new("Failed to execute ssh command '$cmd', exit code: $LastExitCode")
```

Comment:
For the lazy, you can do a simple throw with a string.

Reply:
Nice to know!